HIP/hip-0000-long-term-scheduled-transactions.md

```diff
@@ -0,0 +1,399 @@
+    - Get all the transactions already scheduled for the second that the `expirationTime` falls in.
+    - Send all the transactions plus the new transaction through `scheduleThrottles`
+      - if the throttles are exceeded, reject the transaction
+      - Add a `SCHEDULE_FUTURE_THROTTLE_EXCEEDED` error code for this case.
+    - Add up the `gasLimit` for all the transactions plus the new transaction.
+      - if the total gas limit is greater than `contracts.scheduleThrottleMaxGasLimit`, reject the transaction
+      - Add a `SCHEDULE_FUTURE_GAS_LIMIT_EXCEEDED` error code for this case.
+      - NOTE: like throttling at the GRPC level, `gasLimit` is the max amount of gas the user specified in the transaction, not the actual gas it uses.
+  - Throttling cannot happen in pre-check, it shall happen in `handleTransaction`.
+    - ie `ScheduleCreate` transactions can fail after they are submitted due to throttles.
+  - The throttles used for limiting transactions inside `handleTransaction` shall be the existing throttles plus the `scheduleThrottles`.
+    - **When planning capacity, Hedera should consider up to `scheduling.maxTxnPerSecond` extra transactions being executed per second.**
+  - The gas throttle setting inside `handleTransaction` shall remain the `contracts.consensusThrottleMaxGasLimit` value.
+    - **Hedera should consider `contracts.schedulingThrottleMaxGasLimit` plus `contracts.frontendThrottleMaxGasLimit` when setting the `contracts.consensusThrottleMaxGasLimit` value.**
+
+### handleTransaction logic
+
+- During `handleTransaction`, after the "current" transaction executes, Scheduled Transactions with an `expirationTime` **less than** the current second shall be executed or expired.
+  - The number of Scheduled Transactions that execute shall be up to 999 per `handleTransaction` call.
+    - This is based on there being up to 999 extra consensus timestamps available per call to `handleTransaction`.
+    - A smaller number of transactions shall execute if the "current" transaction causes other synthetic transactions to process before we execute Scheduled Transactions.
+
+### Required Signatures
+
+- If sufficient signatures are not received before the `expirationTime`, Scheduled Transactions shall expire when they are evaluated, without executing, in all cases.
+
+### Fees
+
+- `ScheduleCreate` containing `ContractCall` transactions shall have a fee of $0.10.
+  - `ScheduleCreate` shall remain at a $0.01 fee for all other transaction types.
+  - **Fees will still be dynamic based on resource usage.**
+  
+### Transaction Types
+
+- The `scheduling.whitelist` setting shall be updated to - `ConsensusSubmitMessage,CryptoTransfer,TokenMint,TokenBurn,CryptoCreate,CryptoUpdate,FileUpdate,SystemDelete,SystemUndelete,Freeze,ContractCall,ContractCreate,ContractUpdate,ContractDelete`
+
+### Mirror Node Change
+
+- Add `waitForExpiry` to the mirror node schedule api.
```

Comment:
We should also add `expirationTime` to the API.

Reply:
It’s already there.

Reply:
This is the JSON output of `/api/v1/schedules/{id}`. I don't see `expiration_time`, which is a new field you added in this HIP:

```json
{
   "admin_key":null,
   "deleted":false,
   "consensus_timestamp":"1620327749.254652999",
   "creator_account_id":"0.0.950",
   "executed_timestamp":"1620327749.254653000",
   "memo":"",
   "payer_account_id":"0.0.950",
   "schedule_id":"0.0.226412",
   "signatures":[
      {
         "consensus_timestamp":"1620327749.254652999",
         "public_key_prefix":"Qg==",
         "signature":"SUZHg7uixmb5MjubWNSs4H1FRQdgHwgj9KEe0XWQD4yvD9W+AAQkZBy+FdOnYbz/bfTT3op++Opg\nfGKKRqYKAg==",
         "type":"ED25519"
      }
   ],
   "transaction_body":"CIDC1y8SFFNob2NrZWQsIEkgdGVsbCB5b3UhShMKEQoGCgIYYhACCgcKAxi2BxAB"
}
```

Reply:
You are correct, sorry. Will add. I have been staring at the services code too much.